plan/physical.go

```diff
@@ -70,6 +58,16 @@ func (pp *physicalPlanner) Plan(spec *Spec) (*Spec, error) {
 		}
 	}
 
+	// Compute time bounds for nodes in the plan
+	if err := transformedSpec.BottomUpWalk(ComputeBounds); err != nil {
```

Comment:
I'm not sure about setting the trigger specs, but wouldn't you want to propagate time ranges before doing validation?

I think this is fine for now, but the complexity of how the different passes interact will increase over time... Intuitively it seems like validation should be last, since you wouldn't want to validate and then do something that invalidates the plan; in that case we wouldn't be able to catch it.

Reply:
You're right. Perhaps the mechanism for setting trigger specs needs to change?

Basically trigger specs are a property of physical plan nodes. However, if a range is not pushed down, then the output of physical planning is a plan with a logical `FromProcedureSpec`. So because we set the trigger specs first, we try to set a trigger spec on the logical `From` node, but we error because it should be a physical plan node.

Reply:
Maybe make the trigger-setting pass be more robust and do nothing for logical nodes?  We know that validation will catch them and produce a reasonable error.